proposal/indented-syntax-improvements.md

```diff
@@ -0,0 +1,244 @@
+(1, 2,\n 3)` can be parsed.
+
+Existing syntax allows for trailing commas in Sass lists. However, comma
+separated lists can not use a trailing comma to signify that a list will
+continue after the line break, as this would break existing stylesheets with
+trailing commas.
+
+Because arguments to functions and mixins are already wrapped in `()`, line
+breaks in arguments do not need to cause a statement break. Interpolations are
+wrapped in `#{}` so line breaks do not need to end statements.
+
+#### At-rules
+
+For any at-rule that is supported by native CSS, line breaks after the `@` and
+before a block or statement are not supported. An exception is that line breaks
+are supported inside parentheses. This includes
+`@import`, (which overlaps with Sass), `@supports`, `@media`, `@keyframes` and
+any unknown at-rule.
+
+These rules should be emitted as is, with no special handling from Sass,
+including all parentheses.
+
+### Design Decisions
+
+While some CSS at-rules may have contexts where a line break would not be
+meaningful, custom handling of line breaks is outside of the scope of this
+proposal. For instance, `@media (hover: hover) and \n (color)` is not supported,
+even though line breaks do not end statements after boolean operators in general
+SassScript. This avoids a requirement for Sass to add custom support for new CSS
+at-rules.
+
+We considered borrowing alternate syntax from other languages, such as a leading
+`>` or `|` from YAML or a trailing `|` from HAML. These introduce syntax that is
+novel to Sass, and we instead opted to borrow syntax from the SCSS format. It
+also could introduce incompatibilities with future CSS features.
+
+This proposal introduces the changes for all authors using the indented syntax,
+as opposed to introducing a separate syntax or compiler flag. As the proposal is
+addititive to existing syntax, authors can choose to not use the new syntax, and
+can choose to limit the syntax with linters.
+
+Instead of just allowing optional curly braces around indented syntax, we opted
+to require SCSS syntax inside of the braces. This is intentionally
+unidirectional to prevent issues with multiple nested formats.
+
+## Syntax
+
+### IndentedStatements
+
+<x><pre>
+**IndentedStatements**  ::= (Statement ';'? [IndentSame])\* Statement ';'?
+</pre></x>
+
+[IndentSame]: ../spec/statement.md#indentation
+
+Remove the following prose:
+
+The `Statement` productions may not include newlines outside of `IndentSame`
+productions.
+
+### Block
+
+Replace footnote 1 with:
+
+1: In the SCSS syntax, only the `ScssBlock` production is valid.
+
+### WhitespaceComment
+
+In footnote 2, remove:
+
+In the indented syntax, this may not contain newlines.
+
+### Whitespace
+
+Replace footnote 1 with:
+
+1. In the indented syntax, [`LineBreak`] is not whitespace in the
+   `ImportAtRule`, `SupportsAtRule`, [`MediaAtRule`], `KeyframesAtRule`, or
+   [`UnknownAtRule`], except inside of parentheses or square brackets as defined
```

Comment:
You don't need to special-case `UnknownAtRule` here because statement breaks are allowed essentially anywhere outside of matched braces anyway, so the existing clause will already work. You don't need to special-case `KeyframesAtRule` because it's actually parsed as an `UnknownAtRule` and only recognized as `@keyframes` during evaluation.